corehq/middleware.py

```diff
@@ -195,3 +204,43 @@ def process_view(self, request, view_func, view_args, view_kwargs):
             client.tags_context({
                 'domain': request.domain
             })
+
+
+class SessionBypassMiddleware(SessionMiddleware):
+    def __init__(self, get_response=None):
+        super(SessionBypassMiddleware, self).__init__(get_response)
+        regexes = getattr(settings, 'SESSION_BYPASS_URLS', [])
+        self.bypass_re = [
+            re.compile(regex.format(domain=legacy_domain_re)) for regex in regexes
+        ]
+
+    def process_request(self, request):
+        if self._bypass_sessions(request):
+            return
+
+        super(SessionBypassMiddleware, self).process_request(request)
+
+    def process_response(self, request, response):
+        if self._bypass_sessions(request):
```

Comment:
Thanks @emord. Option 1 is similar to something on the SO post where you set the save function to a noop. 
```
request.session.save = lambda: None
```

I think doing something like this is likely the best option since it still allows access to the session object but prevents the hit on the cache. The one thing I don't like is that if the session IS being used somewhere we wouldn't know about it. 

The other thing I was thinking of was wrapping the session in a proxy object that logs access so that we can see where it's being used. Doing that would allow us to see if it would be safe to not set the session on the request at all.